tests/saitests/py3/sai_qos_tests.py

```diff
@@ -997,20 +997,23 @@ def runTest(self):
             # allow enough time for the dut to sync up the counter values in counters_db
             time.sleep(8)
 
-            if ('arista' in hwsku.lower() and 'broadcom' in asic_type.lower()) or hwsku == 'DellEMC-Z9332f-M-O16C64' or hwsku == 'DellEMC-Z9332f-O32':
-                xmit_counters, queue_counters = sai_thrift_read_port_counters(
-                    self.client, port_list[dst_port_id])
-                actual_pkts_num_leak_out = xmit_counters[TRANSMITTED_PKTS] - \
-                    xmit_counters_base[TRANSMITTED_PKTS]
-                send_packet(self, src_port_id, pkt, actual_pkts_num_leak_out)
-                sys.stderr.write('Send {} packets as leakout compensation'.format(actual_pkts_num_leak_out))
-                time.sleep(8)
+            if 'broadcom' in asic_type.lower() or hwsku == 'DellEMC-Z9332f-M-O16C64' or hwsku == 'DellEMC-Z9332f-O32':
+                xmit_counters, _ = sai_thrift_read_port_counters(self.client, port_list[dst_port_id])
+                counters_base = xmit_counters_base
+                actual_pkts_num_leak_out = xmit_counters[TRANSMITTED_PKTS] - counters_base[TRANSMITTED_PKTS]
+                while actual_pkts_num_leak_out > 0:
```

Comment:
Should we add wait_until here to set a timeout for this?

Reply:
add retry limitation.